thoth/user_api/api_v1.py

```diff
@@ -414,6 +414,38 @@ def list_python_package_indexes():
     return graph.python_package_index_listing()
 
 
+def list_hardware_environments(
+    is_external: bool = False,
+    page: int = 0,
+):
+    """List hardware environments in the graph database."""
+    graph = GraphDatabase()
+    graph.connect()
+    return {
+        "parameters": {
+            "is_external": False,
+            "page": page,
+        },
+        "hardware_environments": graph.get_hardware_environments(is_external, page),
+    }
+
+
+def list_software_environments(
+    is_external: bool = False,
+    page: int = 0,
+):
+    """List software environments in the graph database."""
+    graph = GraphDatabase()
+    graph.connect()
+    return {
+        "parameters": {
+            "is_external": False,
```

Comment:
Please remove `is_external` from parameters completely (in the `openapi.yaml` specification as well) and call the method as:

```python
graph.get_software_environments(is_external=False, offset=page)
```

Reply:
Thanks!